### PR TITLE
sql: use DelRange with tombstone in `force_delete_table_data`

### LINKED ISSUE #85754: sql: `crdb_internal.force_delete_table_data` should use `DeleteRange` in 22.2
**Is your feature request related to a problem? Please describe.**
Relates to https://github.com/cockroachdb/cockroach/issues/70427. Maybe the system tenant should be able to issue `ClearRange` optionally, but it's not totally clear. The secondary tenants shouldnot.

**Describe the solution you'd like**

Change the `ClearRange` code after the version is active to use `DeleteRange` with tombstones.

Jira issue: CRDB-18423

Epic CRDB-2624

### PR DESCRIPTION
**Pull request by ajwerner**
Fixes #85754

Release justification: minor change needed to adopt MVCC bulk ops fully

Release note: None

### CODE REVIEW

File: pkg/sql/repair.go [hunk 1]
```diff
@@ -780,17 +781,22 @@ func (p *planner) ForceDeleteTableData(ctx context.Context, descID int64) error
...
-		},
-	})
-
-	err = p.txn.DB().Run(ctx, b)
-	if err != nil {
+	if p.execCfg.Settings.Version.IsActive(ctx, clusterversion.UseDelRangeInGCJob) {
+		b.AddRawRequest(&roachpb.DeleteRangeRequest{
+			RequestHeader:     requestHeader,
+			UseRangeTombstone: true,
+		})
```
Comment (erikgrinaker) [inline pkg/sql/repair.go:793]: Do we need to worry about unbounded parallelization here? Also, is it worth adding in admission control headers?

Comment (ajwerner) [reply pkg/sql/repair.go:793]: We didn't in the `ClearRange` code which preceded this. I think we have all the same fast-paths now, no?

Comment (ajwerner) [reply pkg/sql/repair.go:793]: I guess the argument is that the user has decided to take this dire action. I don't know. I honestly have never used this thing. It came out of a post-mortem. I don't care about it really.

Comment (erikgrinaker) [reply pkg/sql/repair.go:793]: Hm ok, probably fine then. Seems like the sort of thing you'd pull in an emergency.

**Review by erikgrinaker — APPROVED** [review]

**Comment by ajwerner** [comment]
TFTR!

bors r+

**Pull request merged**